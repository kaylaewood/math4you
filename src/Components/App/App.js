import React from 'react';
import { Route } from 'react-router-dom';
import HomePage from '../HomePage/HomePage';
import NextButton from '../NextButton/NextButton';
import GameForm from '../GameForm/GameForm';
import Header from '../Header/Header';
import MathBoard from '../MathBoard/MathBoard';
import EndPage from '../EndPage/EndPage';
import './App.scss';

function App() {
  return (
    <main>
      <Route exact path = '/'>
        <HomePage />
        <NextButton nextLink = '/select'/>
      </Route>
      <Route exact path = '/select'>
        <GameForm />
        <NextButton nextLink = '/play'/>
      </Route>
      <Route exact path = '/play'>
        <Header />
        <MathBoard />
      </Route>
      <Route exact path = '/end'>
        <EndPage />
        <NextButton nextLink = '/'/>
      </Route>
    </main>
  );
}

export default App;
