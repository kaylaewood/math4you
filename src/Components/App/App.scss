@import '../../variables.scss';

* {
  font-family: $primary-font;
}

h1 {
  font-family: $accent-font;
}

main {
  height: 100vw;
  background-image: url('../../images/background-img.png');
  background-position: center;
  background-repeat: repeat;
  background-size: contain;
  display: flex;
}

.game-board {
  flex-direction: column;
}
