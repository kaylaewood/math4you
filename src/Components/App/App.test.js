import React from 'react';
import ReactDOM from 'react-dom';
import { App, mapStateToProps } from './App';
import { shallow } from 'enzyme';

describe('App', () => {

  it('should match the snapshot', () => {
    let wrapper = shallow(<App timerIsOver={true}/>)

    expect(wrapper).toMatchSnapshot();
  });

  it('should match the snapshot', () => {
    let wrapper = shallow(<App timerIsOver={false}/>)

    expect(wrapper).toMatchSnapshot();
  });

  describe('mapStateToProps', () => {
    it('should return an object with boolean', () => {

      const mockState = {
        currentProblemSet: 'simplify',
        timer: {
          startTime: 30,
          isOver: true
        },
        gameStats: {
          numberCorrect: 3,
          numberIncorrect: 2
        }
      };

      const expected = {
        timerIsOver: true
      };

      const mappedProps = mapStateToProps(mockState);

      expect(mappedProps).toEqual(expected);
    });
  });

});
