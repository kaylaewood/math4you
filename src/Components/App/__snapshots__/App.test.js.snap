// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot 1`] = `ShallowWrapper {}`;

exports[`App should match the snapshot 2`] = `ShallowWrapper {}`;
